BrainPortal/cbrain_plugins/cbrain-plugins-base/userfiles/xml_file/views/_xml_raw.html.erb

```diff
@@ -5,5 +5,5 @@ Some modern browsers have a built-in XML viewer.
 <p>
 
 <%= link_to "Browser-based viewer for XML content",
-    content_userfile_path(@userfile.id, :disposition => 'inline'), :target => :_blank %>
+    content_userfile_path(@userfile.id, :disposition => 'inline', :file_name => params[:file_name]), :target => :_blank %>
```

Comment:
Again, no such parameter :file_name in the content action

Reply:
Never mind, it's in the userfile_for_viewer() helper

Reply:
But please, the reason we build @userfile in the controller is to encapsulate and validate the information about what is being viewed, don't use `params[:file_name]` (which is PRE validation) and use @userfile.name instead!